### ISSUE TITLE
Using FadingActionBar library as a fragment

### ISSUE THREAD

**Issue by h0aX14** [neokree/MaterialNavigationDrawer#78]
Can anyone help me to implement this https://github.com/ManuelPeinado/FadingActionBar as a fragment with this library ?

**Comment by mkosik** [comment]
Is there any specific issue you need to help with or sth conflicting with this lib, so you cannot implement it? You can get toolbar (as actionbar), set yours and figure all other logic in init() or inside your fragments, don't you?
Anyway, if you will come up with some code, post it here :) It might be valuable for others or we might identify a conflicting parts with this lib.

**Comment by h0aX14** [comment]
@mkosik , I tried all the combinations. I need that in my fragment. I'm posting two codes. Only one of them works and that's for ActionBarActivity but it's not a good practice to start a activity on section clicked..

Here is a Fragment code segment with that library. I have tried setting the theme with different ways, nothing works out. Even if I remove all theme codes in all methods, there is no change.

    public class SampleFragment extends Fragment {
    private FadingActionBarHelper mFadingHelper;
    private Bundle mArguments;

    public static final String ARG_IMAGE_RES = "image_source";
    public static final String ARG_ACTION_BG_RES = "image_action_bs_res";


    @Override
    public void onCreate(Bundle savedInstanceState) {
        getActivity().setTheme(R.style.AppTheme_Light_TranslucentActionBar);
        super.onCreate(savedInstanceState);

    }

    @Override

    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = mFadingHelper.createView(inflater);
      // create ContextThemeWrapper from the original Activity Context with the custom theme
        final Context contextThemeWrapper = new ContextThemeWrapper(getActivity(), R.style.AppTheme_Light_TranslucentActionBar);

        // clone the inflater using the ContextThemeWrapper
        LayoutInflater localInflater = inflater.cloneInContext(contextThemeWrapper);
        if (mArguments != null){
            ImageView img = (ImageView) view.findViewById(R.id.image_header);
            img.setImageResource(mArguments.getInt(ARG_IMAGE_RES));
        }

        return view;
    }

    @Override
    public void onAttach(Activity activity) {
        activity.setTheme(R.style.AppTheme_Light_TranslucentActionBar);

        super.onAttach(activity);

        mArguments = getArguments();
        int actionBarBg = mArguments != null ? mArguments.getInt(ARG_ACTION_BG_RES) : R.drawable.ab_background_indigo;

        mFadingHelper = new FadingActionBarHelper()
                .actionBarBackground(actionBarBg)
                .headerLayout(R.layout.header_light)
                .contentLayout(R.layout.activity_scrollview)
                .lightActionBar(actionBarBg == R.drawable.ab_background_indigo);

        mFadingHelper.initActionBar(activity);
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
       // if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.KITKAT) {
           getActivity().setTheme(R.style.AppTheme_Light_TranslucentActionBar);
        //}
        super.onActivityCreated(savedInstanceState);


    }


Here is a working Activity subclass which I don't need. I created it just to check whether library is actually working out or not with this one.

     public class SampleActivity extends ActionBarActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        FadingActionBarHelper helper = new FadingActionBarHelper()
                .actionBarBackground(R.drawable.ab_background_indigo)
                .headerLayout(R.layout.header_light)
                .contentLayout(R.layout.activity_scrollview)
                .headerOverlayLayout(R.layout.header_overlay);
        setContentView(helper.createView(this));
        helper.initActionBar(this);
    }
    }

**Comment by phillwiggins** [comment]
I think its an issue with calling ActionBarActivity. I seem to be having trouble with SetSupportActionBar? I am getting a cannot resolve error. I think this looks similar?

**Comment by phillwiggins** [comment]
https://mdxdave.de/android/android-fading-actionbar

**Comment by h0aX14** [comment]
@phillwiggins , does the above link works ?

**Comment by phillwiggins** [comment]
Yes, I used Google Translate read the instructions. Had it setup in 5 minutes. He has a GitHub with an example on too which is easy to use.

**Comment by h0aX14** [comment]
@phillwiggins

Can you help me implement that within a Fragment? 
Sorry for late reply, I was away

**Comment by phillwiggins** [comment]
@h0aX14 - Copy and play with this

Fragment layout xml

    <FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
       xmlns:app="http://schemas.android.com/apk/res-auto"
        android:layout_width="match_parent"
        android:layout_height="match_parent">

    <com.purewow.myapplication.ScrollViewHelper
        android:id="@+id/scrollViewHelper"
        android:layout_width="match_parent"
        android:layout_height="wrap_content">

        <LinearLayout
            android:orientation="vertical"
            android:layout_width="match_parent"
            android:layout_height="wrap_content">

            <ImageView
                android:layout_width="match_parent"
                android:layout_height="200dp"
                android:src="@drawable/london"
                android:scaleType="centerCrop" />

            <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
                android:layout_width="fill_parent"
                android:layout_height="80dp"
                android:orientation="horizontal"
                android:background="#000000">



            </LinearLayout>

        </LinearLayout>
    </com.purewow.myapplication.ScrollViewHelper>

    <android.support.v7.widget.Toolbar
        android:id="@+id/toolbar"
        android:layout_width="match_parent"
        android:layout_height="71sp"
        app:theme="@style/ThemeOverlay.AppCompat.Dark.ActionBar"
        android:minHeight="?attr/actionBarSize"
        android:background="?attr/colorPrimary" />

    </FrameLayout>

Fragment.Java

    import android.graphics.drawable.ColorDrawable;
    import android.os.Bundle;
    import android.support.v7.app.ActionBar;
    import android.support.v7.app.ActionBarActivity;
    import android.support.v7.widget.Toolbar;
    import android.text.Spannable;
    import android.text.SpannableString;

       public class Recipe extends ActionBarActivity {

    /** Alpha Toolbar **/
    private com.purewow.myapplication.AlphaForeGroundColorSpan mAlphaForegroundColorSpan;
    private SpannableString mSpannableString;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.recipe);

        String title = "Recipe";

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        final ColorDrawable cd = new ColorDrawable(getResources().getColor(R.color.colorPrimary));

        ActionBar actionBar = getSupportActionBar();
        actionBar.setDisplayHomeAsUpEnabled(true);
        actionBar.setTitle("");
        actionBar.setBackgroundDrawable(cd);

        cd.setAlpha(1);

        com.purewow.myapplication.ScrollViewHelper scrollViewHelper = (com.purewow.myapplication.ScrollViewHelper)findViewById(R.id.scrollViewHelper);
        scrollViewHelper.setOnScrollViewListener(new com.purewow.myapplication.ScrollViewHelper.OnScrollViewListener() {

            @Override
            public void onScrollChanged(com.purewow.myapplication.ScrollViewHelper v, int l, int t, int oldl, int oldt) {
                setTitleAlpha(255 - getAlphaforActionBar(v.getScrollY()));
                cd.setAlpha(getAlphaforActionBar(v.getScrollY()));
            }

            private int getAlphaforActionBar(int scrollY) {
                int minDist = 0, maxDist = 550;
                if(scrollY>maxDist){
                    return 255; }
                else {
                    if (scrollY < minDist) {
                        return 0;
                    } else {
                        return (int) ((255.0 / maxDist) * scrollY);
                    }
                }
            }
        });

        mSpannableString = new SpannableString(title);
        mAlphaForegroundColorSpan = new com.purewow.myapplication.AlphaForeGroundColorSpan(0xFFFFFF);
    }

    private void setTitleAlpha(float alpha) {
        if(alpha<1){ alpha = 1; }
        mAlphaForegroundColorSpan.setAlpha(alpha);
        mSpannableString.setSpan(mAlphaForegroundColorSpan, 0, mSpannableString.length(), Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
        getSupportActionBar().setTitle(mSpannableString);
    }


    }

ScrollViewHelper.Java

    import android.content.Context;
    import android.util.AttributeSet;
    import android.widget.ScrollView;

    public class ScrollViewHelper extends ScrollView {

    private OnScrollViewListener mOnScrollViewListener;

    public ScrollViewHelper(Context context) {
        super(context);
    }
    public ScrollViewHelper(Context context, AttributeSet attrs) {
        super(context, attrs);
    }
    public ScrollViewHelper(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    public interface OnScrollViewListener {
        void onScrollChanged(ScrollViewHelper v, int l, int t, int oldl, int oldt );
    }

    public void setOnScrollViewListener(OnScrollViewListener l) {
        this.mOnScrollViewListener = l;
    }

    protected void onScrollChanged(int l, int t, int oldl, int oldt) {
        mOnScrollViewListener.onScrollChanged( this, l, t, oldl, oldt );
        super.onScrollChanged( l, t, oldl, oldt );
    }
    }

AlphaForeGroundColorSpan.java

    import android.graphics.Color;
    import android.text.TextPaint;
    import android.text.style.ForegroundColorSpan;

    public class AlphaForeGroundColorSpan extends ForegroundColorSpan {
    private float mAlpha;

    public AlphaForeGroundColorSpan(int color) {
        super(color);
    }

    @Override
    public void updateDrawState(TextPaint textPaint) {
        textPaint.setColor(getAlphaColor());
    }

    public void setAlpha(float alpha) {
        mAlpha = alpha;
    }

    private int getAlphaColor() {
        int foregroundColor = getForegroundColor();
        return Color.argb((int) (mAlpha * 255), Color.red(foregroundColor), Color.green(foregroundColor), Color.blue(foregroundColor));
    }
    }

**Comment by eirinaios** [comment]
@phillwiggins
Thanks for pointing that out! I did play a little bit with it. The problem arises from the fact that in the code you provide, class Recipe which lives inside Fragment.java is an Activity and not a Fragment! After looking at the source code of MaterialNavigationDrawer.java, I realized that extensive modifications need to be done in it, so that it can support such an effect for the toolbar, since the layout needs to be changed completely, and include in it a ScrollView for the entire contents area. The problem with that (I tried it) is that the section fragments cannot contain another scrollview in them!

@neokree
Will it be possible to support such a functionality in a future release?
